### DISCUSSION TITLE
[@types/ckeditor__ckeditor5-link] LinkConfig decorators

### DISCUSSION THREAD

**Discussion by knightcode** [DefinitelyTyped/DefinitelyTyped#60303] [Issues with a @types package]
In `link.d.ts`, according to [this](https://ckeditor.com/docs/ckeditor5/latest/features/link.html#demo-2) documentation and run time errors, the decorators property below is incorrect. This:
```typescript
export interface LinkConfig {
    addTargetToExternalLinks?: boolean | undefined;
    decorators?: Record<string, LinkDecoratorDefinition> | undefined;
    defaultProtocol?: string | undefined;
}
```

should be something like:
```typescript
export interface LinkConfig {
    addTargetToExternalLinks?: boolean | undefined;
    decorators?: Array<LinkDecoratorAutomaticDefinition | LinkDecoratorManualDefinition> | undefined
    defaultProtocol?: string | undefined;
}
```

[a comment by an automated account is not shown]

**Comment by fedemp** [comment]
@knightcode IMO docs are wrong in this case. 

Check `linkEditing.js`, line 98: https://github.com/ckeditor/ckeditor5/blob/66b3309aac772c8f976244b9629a853242af7b9d/packages/ckeditor5-link/src/linkediting.js#L98

```js
const linkDecorators = getLocalizedDecorators( editor.t, normalizeDecorators( editor.config.get( 'link.decorators' ) ) );
```

and then, `normalizeDecorators` at `utils.js`, https://github.com/ckeditor/ckeditor5/blob/66b3309aac772c8f976244b9629a853242af7b9d/packages/ckeditor5-link/src/utils.js#L116-L120

```js
export function normalizeDecorators( decorators ) {
	const retArray = [];

	if ( decorators ) {
		for ( const [ key, value ] of Object.entries( decorators ) ) {
```

I will change the `decorators` line to: `decorators?: Record<string, LinkDecoratorManualDefinition | LinkDecoratorAutomaticDefinition> | undefined;` tough.
